src/lib/components/__tests__/Details.svelte.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { render } from '@testing-library/svelte'
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import Details from '../Details.svelte'
+import * as useSelection from '$lib/hooks/useSelection.svelte'
+import * as useWeblabs from '$lib/hooks/useWeblabs.svelte'
+import { Weblab } from '$lib/hooks/useWeblabs.svelte'
+
+describe('Details component', () => {
+	const mockedWeblab = new Weblab()
+
+	beforeEach(() => {
+		// Mock the selection hooks to return test data
+		vi.mocked(useSelection.useFocusedObject).mockReturnValue({
+			current: {
+				name: 'Test Object',
+				uuid: '1234-5678',
+				referenceFrame: 'parent_frame',
+				pose: {
+					x: 10,
+					y: 20,
+					z: 30,
+					oX: 0.1,
+					oY: 0.2,
+					oZ: 0.3,
+					theta: 0.4,
+				},
+				geometry: {
+					case: 'box',
+					value: {
+						dimsMm: { x: 10, y: 20, z: 30 },
+					},
+				},
+				metadata: { batched: undefined },
+			},
+		})
+		vi.mocked(useSelection.useFocusedObject3d).mockReturnValue({
+			current: undefined,
+		})
+		vi.mocked(useWeblabs.useWeblabs).mockReturnValue({
+			weblab: mockedWeblab,
+		})
+	})
+
+	function getDefaultProps(overrides = {}) {
+		return {
+			...overrides,
+		}
+	}
+
+	it('renders object name', () => {
+		const { getByText } = render(Details, getDefaultProps())
+		expect(getByText('Test Object')).toBeDefined()
+	})
+
+	it('renders local details under weblab active', () => {
+		mockedWeblab.isActive = vi.fn(() => true)
+		const { getByText, getByTestId } = render(Details, getDefaultProps())
+
+		expect(getByText('parent frame')).toBeDefined()
```

Comment:
```suggestion
		expect(getByText('parent frame')).toBeInTheDocument()
```
^^ this is our more idiomatic check for dom elements

Reply:
updated